cmake/llext-edk.cmake

```diff
@@ -0,0 +1,149 @@
+        cmake_path(GET flag FILENAME name)
+        relative_dir(${parent} dest bindir)
+        cmake_path(RELATIVE_PATH dest BASE_DIRECTORY ${LLEXT_EDK} OUTPUT_VARIABLE dest_rel)
+        if(bindir)
+            list(APPEND imacros_gen_make "-imacros\$(${install_dir_var})/${dest_rel}/${name}")
+            list(APPEND imacros_gen_cmake "-imacros\${${install_dir_var}}/${dest_rel}/${name}")
+        else()
+            list(APPEND imacros_make "-imacros\$(${install_dir_var})/${dest_rel}/${name}")
+            list(APPEND imacros_cmake "-imacros\${${install_dir_var}}/${dest_rel}/${name}")
+        endif()
+    else()
+        list(APPEND new_cflags ${flag})
+    endif()
+endforeach()
+set(LLEXT_CFLAGS ${new_cflags})
+
+cmake_path(CONVERT "${INTERFACE_INCLUDE_DIRECTORIES}" TO_CMAKE_PATH_LIST INCLUDE_DIRS)
+
+set(autoconf_h_edk ${LLEXT_EDK_INC}/${AUTOCONF_H})
+cmake_path(RELATIVE_PATH AUTOCONF_H BASE_DIRECTORY ${PROJECT_BINARY_DIR} OUTPUT_VARIABLE autoconf_h_rel)
+
+list(APPEND base_flags_make ${LLEXT_CFLAGS} ${imacros_make})
+list(APPEND base_flags_cmake ${LLEXT_CFLAGS} ${imacros_cmake})
+
+file(MAKE_DIRECTORY ${LLEXT_EDK_INC})
+foreach(dir ${INCLUDE_DIRS})
+    if (NOT EXISTS ${dir})
+        continue()
+    endif()
+
+    relative_dir(${dir} dest bindir)
+    # Use destination parent, as the last part of the source directory is copied as well
+    cmake_path(GET dest PARENT_PATH dest_p)
+
+    file(MAKE_DIRECTORY ${dest_p})
+    file(COPY ${dir} DESTINATION ${dest_p} FILES_MATCHING PATTERN "*.h")
+
+    cmake_path(RELATIVE_PATH dest BASE_DIRECTORY ${LLEXT_EDK} OUTPUT_VARIABLE dest_rel)
+    if(bindir)
+        list(APPEND inc_gen_flags_make "-I\$(${install_dir_var})/${dest_rel}")
+        list(APPEND inc_gen_flags_cmake "-I\${${install_dir_var}}/${dest_rel}")
+    else()
+        list(APPEND inc_flags_make "-I\$(${install_dir_var})/${dest_rel}")
+        list(APPEND inc_flags_cmake "-I\${${install_dir_var}}/${dest_rel}")
+    endif()
+endforeach()
+
+if(CONFIG_LLEXT_EDK_USERSPACE_ONLY)
+    # Copy syscall headers from edk directory, as they were regenerated there.
+    file(COPY ${PROJECT_BINARY_DIR}/edk/include/generated/ DESTINATION ${LLEXT_EDK_INC}/zephyr/include/generated)
+endif()
+
+# Generate flags for Makefile
+list(APPEND all_flags_make ${base_flags_make} ${imacros_gen_make} ${inc_gen_flags_make} ${inc_flags_make})
+list(JOIN all_flags_make " " all_flags_str)
+file(WRITE ${LLEXT_EDK}/Makefile.cflags "LLEXT_CFLAGS = ${all_flags_str}")
+
+list(JOIN inc_flags_make " " inc_flags_str)
+file(APPEND ${LLEXT_EDK}/Makefile.cflags "\n\nLLEXT_INCLUDE_CFLAGS = ${inc_flags_str}")
+
+list(JOIN inc_gen_flags_make " " inc_gen_flags_str)
+file(APPEND ${LLEXT_EDK}/Makefile.cflags "\n\nLLEXT_GENERATED_INCLUDE_CFLAGS = ${inc_gen_flags_str}")
+
+file(APPEND ${LLEXT_EDK}/Makefile.cflags "\n\nLLEXT_BASE_CFLAGS = ${base_flags_make}")
+
+file(APPEND ${LLEXT_EDK}/Makefile.cflags "\n\nLLEXT_GENERATED_IMACROS_CFLAGS = ${imacros_gen_make}")
```

Comment:
Definitely missed a list conversion for these, these do end up as a single word with semicolons in a Makefile.